src/main/java/hana/command/ByeCommand.java

```diff
@@ -0,0 +1,44 @@
+package hana.command;
+
+import hana.HanaException;
+import hana.storage.Storage;
+import hana.tasklist.TaskList;
+import hana.ui.Ui;
+import javafx.animation.PauseTransition;
+import javafx.application.Platform;
+import javafx.util.Duration;
+
+/**
+ * Represents a command to exit the program.
+ * When executed, this command will print a goodbye message and signal the application to exit.
+ */
+public class ByeCommand extends Command {
+
+    /**
+     * Executes the ByeCommand which prints a goodbye message.
+     *
+     * @param taskList The list of tasks.
+     * @param ui The UI object used to interact with the user.
+     * @param storage The storage object to handle reading/writing of tasks.
+     * @throws HanaException If an error occurs during command execution.
+     */
+    @Override
+    public void execute(TaskList taskList, Ui ui, Storage storage) throws HanaException {
+        ui.printBye();
+        PauseTransition pause = new PauseTransition(Duration.seconds(3));
+        pause.setOnFinished(event -> {
+            Platform.exit();
+        });
+        pause.play();
```

Comment:
Is this necessary? May I know what the rationale is for?

Reply:
It creates a delay of 3 seconds before the app terminates so that the user can read the exit message rather than exit immediately after printing the message. It's not really necessary but without the delay, there's no point in printing the message since the user wouldn't have time to read it.